### DISCUSSION TITLE
How does remark-mdx plugin work?

### DISCUSSION THREAD

**Discussion by sunknudsen** [remarkjs/remark#1104] [Q&A]
I, perhaps naively, expected [remark-mdx](https://www.npmjs.com/package/remark-mdx) plugin to works as follows (see https://codesandbox.io/s/nervous-waterfall-h7kdcc).

```tsx
import { MantineProvider } from "@mantine/core";
import ReactMarkdown from "react-markdown";
import remarkMdx from "remark-mdx";

export default function App() {
  return (
    <MantineProvider withGlobalStyles withNormalizeCSS>
      <ReactMarkdown
        children={`import Test from "./test"\n\n<Test>bar</Test>`}
        remarkPlugins={[remarkMdx]}
      />
    </MantineProvider>
  );
}
```

## Expected output:

```html
<div id="root">
  <p>
    <span class="css-1xxe9l9">
      yo
    </span>
  </p>
</div>
```

## Output:

```html
<div id="root">
  import Test from "./test"
  <p>
    <div>
      bar
    </div>
  </p>
</div>
```

Would anyone happen to know what I am missing?

**Comment by wooorm** [comment]
No. It doesn’t work like that at all.

And you are running into an XY problem. Probably a good idea to take a step back and think about what you want to achieve, and supply more information about what you want to do: https://mdxjs.com/community/support/.

MDX already does what you want. If you want MDX, use MDX, you don’t need `react-markdown`. Or use `react-markdown` and don’t use MDX.
It’s like using Rollup and esbuild and webpack. They are alternatives to each other, one is enough.

**Comment by sunknudsen** [comment]
Looking for a way to handle custom HTML tags (or components) such as `<kbd>` or `<video data-src="…" />`… perhaps `remark-mdx` is not what I am looking for.

**Reply by wooorm** [reply]
`react-markdown` can do that (`components`, [Appendix A](https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown)), MDX can do that (also called `components`)

**Reply by sunknudsen** [reply]
Thanks for helping out… [rehype-raw](https://github.com/rehypejs/rehype-raw) with custom components is what I was looking for.
